advice.c

```diff
@@ -20,6 +21,33 @@ int advice_add_embedded_repo = 1;
 int advice_ignored_hook = 1;
 int advice_waiting_for_editor = 1;
 
+static int advice_use_color = -1;
+static char advice_colors[][COLOR_MAXLEN] = {
+	GIT_COLOR_RESET,
+	GIT_COLOR_YELLOW,	/* HINT */
+};
+
+enum color_advice {
+	ADVICE_COLOR_RESET = 0,
+	ADVICE_COLOR_HINT = 1,
+};
+
+static int parse_advise_color_slot(const char *slot)
+{
+	if (!strcasecmp(slot, "reset"))
+		return ADVICE_COLOR_RESET;
+	if (!strcasecmp(slot, "advice"))
+		return ADVICE_COLOR_HINT;
+	return -1;
+}
+
+static const char *advise_get_color(enum color_advice ix)
+{
+	if (want_color(advice_use_color))
```

Comment:
The biggest problem with this is that `want_color()` implicitly looks whether `stdout` is connected to an interactive terminal (as opposed to, say, being redirected to a file or to a different command). We use the values returned by `advice_get_color()` on *`stderr`* instead, though.

I am not quite certain what is the most elegant way to address this. Probably extend the signature of `want_color()` to take a `FILE *`, or an `int` (file descriptor)...